prow/config.yaml

```diff
@@ -868,17 +868,85 @@ presubmits:
           path: /mnt/disks/ssd0
 
   - name: pull-kubernetes-unit
-    agent: jenkins
+    agent: kubernetes
     always_run: true
     context: pull-kubernetes-unit
     rerun_command: "/test pull-kubernetes-unit"
     trigger: "(?m)^/test( all| pull-kubernetes-unit),?(\\s+|$)"
+    spec:
+      containers:
+      - image: gcr.io/k8s-testimages/kube-test
+        args:
+        - "--clean"
+        - "--root=/go/src"
+        - "--job=$(JOB_NAME)"
+        - "--repo=github.com/Huawei-PaaS/$(REPO_NAME)=$(PULL_REFS)"
+        - "--service-account=/etc/service-account/service-account.json"
+        - "--upload=gs://huawei-kubernetes-jenkins/pr-logs"
+        securityContext:
+          privileged: true
+        volumeMounts:
+        - name: service
+          mountPath: /etc/service-account
+          readOnly: true
+        - name: docker
+          mountPath: /var/run/docker.sock
```

Comment:
I don't think we want to mount in the docker socket. See: https://github.com/kubernetes/kubernetes/issues/1806#issuecomment-301530201

Reply:
Correct. We could actually add a unit test preventing any hostPath of `/var/run/docker.sock`.

Reply:
done in https://github.com/kubernetes/test-infra/pull/5000

Reply:
I took a look about the alternatives which it appear to me that DinD would be more reasonable (still security issues but resources are managed in kubernetes).

Reply:
Don't worry about security issues. We run all test pods in an isolated project, so rogue code shouldn't be able to find any important secrets.